BLSQ/iaso
# IA-2900: registry view permission

With this permission, users would not be able to:

- see other Forms than the Reference forms
- edit form submissions via Enketo
- consult the Org unit details
- download data

Therefore the corresponding buttons should be hidden from the interface

Related JIRA tickets : IA-2900

## Self proofreading checklist



## Changes

- adding read and write permissions
- adapt backend to filter instances and get only reference instances
- display /hide buttons regarding permissions
- use `DisplayIfUserHasPerm`
- add strict mode to `DisplayIfUserHasPerm`and create a new `userHasAllPermissions`function

## How to test

You need a non superadmin user having only registry_read rights.
Connect to a registry page with this user and check that buttons are not present

## Print screen / video

![Screenshot 2024-05-22 at 10 46 06](https://github.com/BLSQ/iaso/assets/12494624/8a0a69e4-43dc-42b5-b82e-e7000ab36085)
